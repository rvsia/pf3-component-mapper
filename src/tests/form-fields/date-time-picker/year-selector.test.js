import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import YearSelector from '../../../form-fields/date-time-picker/year-selector';

describe('<YearSelector />', () => {
  let initialProps;
  let initialState;
  let initialYear;

  beforeEach(() => {
    initialYear = 2019;
    initialProps = {
      yearChange: jest.fn(),
      toggleSelectingYear: jest.fn(),
    };
    initialState = {
      initialYear,
      firstInterval: [ initialYear - 19, initialYear ],
      currentInterval: [ initialYear - 19, initialYear ],
    };
  });

  it('should render correctly', () => {
    const wrapper = mount(<YearSelector { ...initialProps } />);
    wrapper.setState({ ...initialState });
    wrapper.update();
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render correctly with some selected date', () => {
    const wrapper = mount(<YearSelector { ...initialProps } selectedDay={ new Date(Date.UTC(2019)) }/>);
    wrapper.setState({ ...initialState });
    wrapper.update();
    expect(wrapper.findWhere(node => node.key() === 'year-cell-4').last().find('button').props().className).toEqual('selected');
  });

  it('should select year 2000', () => {
    const yearChange = jest.fn();
    const wrapper = mount(<YearSelector { ...initialProps } yearChange={ yearChange } />);
    wrapper.setState({ ...initialState });
    wrapper.update();
    wrapper.find('table').last().find('tr').first().find('td').first().find('button').simulate('click');
    expect(yearChange).toHaveBeenCalledWith(2000);
  });

  it('should switch to prev year interval', () => {
    const wrapper = mount(<YearSelector { ...initialProps } />);
    wrapper.setState({ ...initialState });
    wrapper.update();
    wrapper.find('table').first().find('td').first().simulate('click');
    expect(wrapper.state().currentInterval).toEqual([ 1980, 1999 ]);
  });

  it('should witch to next year interval', () => {
    const wrapper = mount(<YearSelector { ...initialProps } />);
    wrapper.setState({ ...initialState });
    wrapper.update();
    wrapper.find('table').first().find('td').last().simulate('click');
    expect(wrapper.state().currentInterval).toEqual([ 2020, 2039 ]);
  });
});
