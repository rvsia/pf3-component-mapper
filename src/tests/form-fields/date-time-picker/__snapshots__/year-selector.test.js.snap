// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<YearSelector /> should render correctly 1`] = `
<YearSelector
  disabledDays={Array []}
  toggleSelectingYear={[MockFunction]}
  yearChange={[MockFunction]}
>
  <div
    className="DayPicker"
  >
    <div
      className="DayPicker-wrapper"
      tabIndex="0"
    >
      <div>
        <YearIntervalSelector
          currentInterval={
            Array [
              2000,
              2019,
            ]
          }
          nextInterval={[Function]}
          prevInterval={[Function]}
        >
          <table
            className="year-interval-header"
          >
            <tbody>
              <tr>
                <td
                  onClick={[Function]}
                >
                  <Icon
                    name="angle-left"
                    type="fa"
                  >
                    <FontAwesome
                      name="angle-left"
                    >
                      <span
                        aria-hidden={true}
                        className="fa fa-angle-left"
                      />
                    </FontAwesome>
                  </Icon>
                </td>
                <td>
                  <button
                    className="year-interval-button"
                    disabled={true}
                  >
                    2000 - 2019
                  </button>
                </td>
                <td
                  onClick={[Function]}
                >
                  <Icon
                    name="angle-right"
                    type="fa"
                  >
                    <FontAwesome
                      name="angle-right"
                    >
                      <span
                        aria-hidden={true}
                        className="fa fa-angle-right"
                      />
                    </FontAwesome>
                  </Icon>
                </td>
              </tr>
            </tbody>
          </table>
        </YearIntervalSelector>
        <table
          className="pf3-datetime-months-table"
        >
          <tbody>
            <tr
              key="year-row-0"
            >
              <td
                className=""
                key="year-cell-0"
              >
                <button
                  className=""
                  onClick={[Function]}
                >
                  2000
                </button>
              </td>
              <td
                className=""
                key="year-cell-1"
              >
                <button
                  className=""
                  onClick={[Function]}
                >
                  2001
                </button>
              </td>
              <td
                className=""
                key="year-cell-2"
              >
                <button
                  className=""
                  onClick={[Function]}
                >
                  2002
                </button>
              </td>
              <td
                className=""
                key="year-cell-3"
              >
                <button
                  className=""
                  onClick={[Function]}
                >
                  2003
                </button>
              </td>
              <td
                className=""
                key="year-cell-4"
              >
                <button
                  className=""
                  onClick={[Function]}
                >
                  2004
                </button>
              </td>
            </tr>
            <tr
              key="year-row-1"
            >
              <td
                className=""
                key="year-cell-0"
              >
                <button
                  className=""
                  onClick={[Function]}
                >
                  2005
                </button>
              </td>
              <td
                className=""
                key="year-cell-1"
              >
                <button
                  className=""
                  onClick={[Function]}
                >
                  2006
                </button>
              </td>
              <td
                className=""
                key="year-cell-2"
              >
                <button
                  className=""
                  onClick={[Function]}
                >
                  2007
                </button>
              </td>
              <td
                className=""
                key="year-cell-3"
              >
                <button
                  className=""
                  onClick={[Function]}
                >
                  2008
                </button>
              </td>
              <td
                className=""
                key="year-cell-4"
              >
                <button
                  className=""
                  onClick={[Function]}
                >
                  2009
                </button>
              </td>
            </tr>
            <tr
              key="year-row-2"
            >
              <td
                className=""
                key="year-cell-0"
              >
                <button
                  className=""
                  onClick={[Function]}
                >
                  2010
                </button>
              </td>
              <td
                className=""
                key="year-cell-1"
              >
                <button
                  className=""
                  onClick={[Function]}
                >
                  2011
                </button>
              </td>
              <td
                className=""
                key="year-cell-2"
              >
                <button
                  className=""
                  onClick={[Function]}
                >
                  2012
                </button>
              </td>
              <td
                className=""
                key="year-cell-3"
              >
                <button
                  className=""
                  onClick={[Function]}
                >
                  2013
                </button>
              </td>
              <td
                className=""
                key="year-cell-4"
              >
                <button
                  className=""
                  onClick={[Function]}
                >
                  2014
                </button>
              </td>
            </tr>
            <tr
              key="year-row-3"
            >
              <td
                className=""
                key="year-cell-0"
              >
                <button
                  className=""
                  onClick={[Function]}
                >
                  2015
                </button>
              </td>
              <td
                className=""
                key="year-cell-1"
              >
                <button
                  className=""
                  onClick={[Function]}
                >
                  2016
                </button>
              </td>
              <td
                className=""
                key="year-cell-2"
              >
                <button
                  className=""
                  onClick={[Function]}
                >
                  2017
                </button>
              </td>
              <td
                className=""
                key="year-cell-3"
              >
                <button
                  className=""
                  onClick={[Function]}
                >
                  2018
                </button>
              </td>
              <td
                className=""
                key="year-cell-4"
              >
                <button
                  className=""
                  onClick={[Function]}
                >
                  2019
                </button>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</YearSelector>
`;
